Framework/Parallel/test/EventParserTest.h

```diff
@@ -0,0 +1,501 @@
+    generateTestData(numBanks, pixelsPerBank, numPulses, maxEventsPerPixel);
+  }
+
+  ~FakeParserDataGenerator() {
+    cleanupEventLists(m_event_lists);
+    for (auto &lists : test_event_lists)
+      cleanupEventLists(lists);
+  }
+
+  const std::vector<int32_t> &bankOffsets() const { return m_bank_offsets; }
+
+  const std::vector<IndexType> &eventIndex(size_t bank) const {
+    return m_event_indices[bank];
+  }
+
+  const std::vector<TimeZeroType> &eventTimeZero() { return m_event_time_zero; }
+
+  const std::vector<TimeOffsetType> &eventTimeOffset(size_t bank) const {
+    return m_event_time_offsets[bank];
+  }
+
+  const std::vector<int32_t> &eventId(size_t bank) const {
+    return m_event_ids[bank];
+  }
+
+  const std::vector<std::vector<TofEvent> *> &eventLists() const {
+    return m_event_lists;
+  }
+
+  LoadRange generateBasicRange(size_t bank) {
+    LoadRange range;
+    range.eventOffset = 0;
+    range.eventCount = m_event_ids[bank].size();
+    range.bankIndex = bank;
+
+    return range;
+  }
+
+  boost::shared_ptr<EventParser<IndexType, TimeZeroType, TimeOffsetType>>
+  generateTestParser() {
+    // create test parser
+    test_event_lists.push_back(prepareEmptyEventLists(m_event_lists.size()));
+    return boost::make_shared<
+        EventParser<IndexType, TimeZeroType, TimeOffsetType>>(
+        std::vector<std::vector<int>>{}, m_bank_offsets,
+        test_event_lists.back());
+  }
+
+  static void
+  cleanupEventLists(std::vector<std::vector<TofEvent> *> &eventLists) {
+    for (auto *list : eventLists) {
+      if (list != nullptr)
+        delete list;
+    }
+  }
+
+private:
+  void generateTestData(const size_t numBanks, const size_t pixelsPerBank,
+                        const size_t numPulses,
+                        const size_t maxEventsPerPixel) {
+    initOffsetsAndIndices(numBanks, numPulses);
+    m_event_time_zero.resize(numPulses);
+    auto numPixels = numBanks * pixelsPerBank;
+
+    m_event_ids.resize(numBanks);
+    m_event_time_offsets.resize(numBanks);
+    std::generate_n(std::back_inserter(m_event_lists), numPixels,
+                    []() { return new std::vector<TofEvent>(); });
+
+    for (size_t pulse = 0; pulse < numPulses; ++pulse) {
+      m_event_time_zero[pulse] = static_cast<TimeZeroType>(pulse * 100000);
+      size_t pulseEventSize = 0;
+      int bank = 0;
+      for (size_t pixel = 0; pixel < numPixels; ++pixel) {
+        auto eventSize = getRandEventSize(1, maxEventsPerPixel / numPulses);
+        pulseEventSize += eventSize;
+        auto *list = m_event_lists[pixel];
+        auto prev_end = list->size();
+        std::generate_n(std::back_inserter(*list), eventSize, [this, pulse]() {
+          return TofEvent(getRandomTimeOffset(100000),
+                          m_event_time_zero[pulse]);
+        });
+        std::fill_n(std::back_inserter(m_event_ids[bank]), eventSize,
+                    static_cast<IndexType>(m_bank_offsets[bank] + pixel));
```

Comment:
If I understand correctly, this is creating events in `event_id` that are sorted by `event_id` (within the pulse and bank)? Sounds like a special case, but probably unlikely to cause issues (maybe distoring performance tests?).